dip-ct.md

```diff
@@ -0,0 +1,166 @@
+
+## Prior Work
+
+There are many different types of CTs. The term originally was used in the context
+of Bitcoin in 2013 and has grown to a research field with many flavors of CTs. Monero
+currently uses RingCT with Bulletproof+ optimizations and the purpose of this DIP is
+to decide exactly which kind of CT is appropriate for Dash. Monero first implemented
+RingCT in 2017 and then added Bulletproofs in 2018 which allow for an 80% reduction
+in size of transactions, reducing blockchain bloat as well as reducing transaction fees.
+A further improvement called Bulletproofs+ was completed in 2022 which further reduces
+transaction size by roughly 5-10% and improves speed by 10%. Yet another improvement
+called Bulletproofs++ is currently being worked on which further reduces transaction
+size and speeds up runtime. Since BP++ is still being actively worked on and has not
+yet been merged into Monero and BP+ are still not widely used it is currently recommended for Dash to use BPs.
+
+## Consensus Protocol Changes
+
+This DIP proposes a new transaction type as well as new address types and therefore is a consensus change.
+
+## Overview
+
+One of the large differences between Dash and Monero is the Elliptic Curve each is based on. Dash uses secp256k1
+which is inherited from Bitcoin Core while Monero uses the Curve25519 curve. Bulletproofs are curve-agnostic, they can be used with any elliptic curve, but when considering exactly how the code will be implemented and which low-level libraries to use, this becomes important. The Monero implementation of Bulletproofs is specific to the elliptic curve they use and can not easily be ported or used in the Dash codebase.
+
+## Details
+
+This DIP documents how Dash can add CTs using Bulletproofs (BPs). The type of CT we propose using has two main parts, a signature and a Bulletproof. The Bulletproof proves that the signature is valid, in particular, that it's values are in between a certain valid range. This prevents an underflow or overflow of values that could be used to subvert the system. Bulletproofs are a type of Non-Interactive Zero Knowledge proof. They prove that the values are valid or invalid without leaking any information about the values themselves.
+
+### Confidential addresses (CT addresses)
+
+New Dash CTs will need a new type of address, a confidential address, to send and receive CT transactions. This means that the Dash full node will require new RPCs to create, validate, list and import confidential addresses. A prefix will need to be decided upon for these addresses so they can easily be differentiated from other Dash addresses. These addresses will be new data in wallet.dat and the code which reads and writes wallet.dat will need to also be updated to store and retrieve this data. The code which rescans blockchain history for transactions belonging to the current wallet will also need to be updated.
```

Comment:
> A prefix will need to be decided upon for these addresses so they can easily be differentiated from other Dash addresses

this DIP should propose one

> The code which rescans blockchain history for transactions belonging to the current wallet will also need to be updated.

How does the logic for detection of utxo relevance change?

Reply:
@PastaPastaPasta latest DIP version takes these 2 review items into account